magenta-lib/src/main/scala/magenta/graph/DeploymentGraph.scala

```diff
@@ -0,0 +1,90 @@
+package magenta.graph
+
+import magenta.tasks.Task
+
+sealed trait Node {
+  def ~>(to: Node) = Edge(this, to)
+  def maybePriority: Option[Int] = None
+}
+case class StartNode() extends Node
```

Comment:
I had some issues with pattern matching on case objects, but turns out those issues don't seem to affect this particular case.